### ISSUE TITLE
discord rpc wrongly reporting non-steam servers

### ISSUE THREAD

**Issue by sapphonie** [tf2classic/tf2classic-issue-tracker#91]
`serverinfo_received` does not report anonymous steamids (sets `steamid` str to `""`) despite them being valid steamids. probably need to hook `player_activate` instead and grab it there. or something.

**Comment by sapphonie** [comment]
wrong! it's because
```
	if ( steamgameserverapicontext->SteamGameServer() &&
		steamgameserverapicontext->SteamGameServer()->BLoggedOn() )
``` in gamerules. duh.

**Comment by sapphonie** [comment]
fixed in 2.0.4
